src/main/java/github/scarsz/discordsrv/listeners/DiscordAccountLinkListener.java

```diff
@@ -42,12 +45,15 @@ public void onPrivateMessageReceived(PrivateMessageReceivedEvent event) {
     }
 
     public void onGuildMemberJoin(GuildMemberJoinEvent event) {
-        // add linked role back to people when they rejoin the server
+        // add linked role and nickname back to people when they rejoin the server
         UUID uuid = DiscordSRV.getPlugin().getAccountLinkManager().getUuid(event.getUser().getId());
         if (uuid != null) {
             Role roleToAdd = DiscordUtil.getRoleByName(event.getMember().getGuild(), DiscordSRV.config().getString("MinecraftDiscordAccountLinkedRoleNameToAddUserTo"));
             if (roleToAdd != null) DiscordUtil.addRoleToMember(event.getMember(), roleToAdd);
             else DiscordSRV.debug("Couldn't add user to null role");
+
+            OfflinePlayer player = Bukkit.getOfflinePlayer(uuid);
+            DiscordSRV.getPlugin().getNicknameUpdater().setNickname(event.getMember(), player);
```

Comment:
This needs a check to ensure `NicknameSynchronizationEnabled` is `true` before setting nicknames